crates/symbolicator-service/src/services/sourcemap_lookup.rs

```diff
@@ -783,14 +791,15 @@ impl ArtifactFetcher {
         if let Some(debug_id) = key.debug_id() {
             let ty = key.as_type();
             for (bundle_uri, bundle) in &self.artifact_bundles {
-                let Ok(bundle) = bundle else { continue; };
+                let Ok((bundle, resolved_with)) = bundle else { continue; };
                 let bundle = bundle.get();
                 if let Ok(Some(descriptor)) = bundle.source_by_debug_id(debug_id, ty) {
                     self.found_via_bundle_debugid += 1;
                     tracing::trace!(?key, "Found file in artifact bundles by debug-id");
                     return Some(CachedFileEntry {
                         uri: CachedFileUri::Bundled(bundle_uri.clone(), key.clone()),
                         entry: CachedFile::from_descriptor(key.abs_path(), descriptor),
+                        resolved_with: *resolved_with,
```

Comment:
just pointing out an edgecase here:

- file A does not have a debug-id, so it queries for bundles and gets an artifact bundle resolved by `release`
- file B does have a debug-id, and is found in the former bundle. Though the bundle is still marked as resolved by `release`.

Reply:
Good catch, fixed.

Reply:
TBH, I don’t know what the right thing to do here is. We found the bundle by `release`, but resolved the file inside the bundle via `debug-id`… So what is the expectation here?

Reply:
My thinking was that we would have found the bundle via debug id if we'd checked this file (with the debug id) first, but that's probably not actually true, right?

Reply:
That is very good reasoning, yes.